extensions/kafka/src/main/java/io/deephaven/kafka/ingest/KafkaIngester.java

```diff
@@ -281,7 +355,9 @@ private void consumerLoop() {
             }
             final long beforePoll = System.nanoTime();
             final long remainingNanos = beforePoll > nextReport ? 0 : (nextReport - beforePoll);
-            boolean more = pollOnce(Duration.ofNanos(remainingNanos));
+            consumerLoopCallbacks.forEach(x -> x.beforePoll(kafkaConsumer));
+            final boolean more = pollOnce(Duration.ofNanos(remainingNanos));
+            consumerLoopCallbacks.forEach(x -> x.afterPoll(kafkaConsumer, more));
```

Comment:
I think you need some error handling here. Do we trust `beforePoll` and `afterPoll` to never throw exceptions? If they do throw, we should probably advise the downstream consumers. We should also handle this and not just hit the thread's UncaughtExceptionHandler.